pinot-core/src/main/java/org/apache/pinot/core/operator/combine/BaseCombineOperator.java

```diff
@@ -75,13 +74,6 @@ public BaseCombineOperator(List<Operator> operators, QueryContext queryContext,
     _futures = new Future[_numThreads];
   }
 
-  public BaseCombineOperator(List<Operator> operators, QueryContext queryContext, ExecutorService executorService,
```

Comment:
I would recommend keeping a different constructor for group-by combine operators and avoid creating the blocking queue because it is not needed. In the future, we might want to change the group-by combine operators to two phases merge like other combine operators, then we can remove the additional constructor

Reply:
done